OloEditor/src/Panels/SceneHierarchyPanel.cpp

```
@@ -44,116 +52,226 @@ namespace OloEngine
+
+                if (hasFilter)
+                {
+                    auto& tag = entity.GetComponent<TagComponent>().Tag;
+                    if (!caseInsensitiveFind(tag, m_FilterText))
+                    {
+                        return;
+                    }
+                }
+                DrawEntityNode(entity); });
 
             if (ImGui::IsMouseDown(0) && ImGui::IsWindowHovered())
```

Clang (14.0.6)

[note] +2, including nesting penalty of 1, nesting level increased to 2

(clang)